src/lib/util/util.mli

```diff
@@ -29,6 +29,7 @@
 (**************************************************************************)
 
 exception Timeout
+exception Step_limit_reached of int
```

Comment:
Those two have a very similar role. We could factor that out into a single `exception Limit_reached of resources_limit` and `type resources_limit = Time | Steps of int` to ensure uniform handling.

(I am fine with keeping the PR as is and making this a long-term cleanup issue instead)

Reply:
I'll do the change right now, I like this factorization

Reply:
(Note that the type should be `resource_limit`, not `resources_limit`, because of English rules being different from French ones)